certification/internal/engine/engine.go

```diff
@@ -0,0 +1,156 @@
+package engine
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path"
+	"time"
+
+	"github.com/google/go-containerregistry/pkg/crane"
+	log "github.com/sirupsen/logrus"
+
+	"github.com/redhat-openshift-ecosystem/openshift-preflight/certification"
+	"github.com/redhat-openshift-ecosystem/openshift-preflight/certification/errors"
+	fileutils "github.com/redhat-openshift-ecosystem/openshift-preflight/certification/internal/utils/file"
+	"github.com/redhat-openshift-ecosystem/openshift-preflight/certification/runtime"
+)
+
+// CraneEngine implements a certification.CheckEngine, and leverage crane to interact with
+// the container registry and target image.
+type CraneEngine struct {
+	// Image is what is being tested, and should contain the
+	// fully addressable path (including registry, namespaces, etc)
+	// to the image
+	Image string
+	// Checks is an array of all checks to be executed against
+	// the image provided.
+	Checks []certification.Check
+	// RegistryCreds are the credentials used to access to registry
+	// from which the image will be pulled.
+	RegistryCreds RegistryCredentials
+	// IsBundle is an indicator that the asset is a bundle.
+	IsBundle bool
+
+	imageRef ImageReference
+	results  runtime.Results
+}
+
+func (c *CraneEngine) ExecuteChecks() error {
+	log.Info("target image: ", c.Image)
+
+	// prepare crane runtime options
+	options := make([]crane.Option, 0)
+
+	// derive authentication context
+	withAuth := AuthConfig(c.RegistryCreds)
+	if withAuth == nil {
+		log.Info("no authentication context derived from execution configuration")
+	} else {
+		log.Info("accessing registry with provided authentication context")
+		options = append(options, *withAuth)
+	}
+
+	// pull the image and save to fs
+	log.Info("pulling image from target registry")
+	img, err := crane.Pull(c.Image, options...)
+	if err != nil {
+		return fmt.Errorf("%w: %s", errors.ErrGetRemoteContainerFailed, err)
+	}
+
+	// create tmpdir to receive extracted fs
+	tmpdir, err := os.MkdirTemp(os.TempDir(), "preflight-*")
+	if err != nil {
+		return fmt.Errorf("%w: %s", errors.ErrCreateTempDir, err)
+	}
+	log.Debug("temporary directory is", tmpdir)
+	defer func() {
+		if err := os.RemoveAll(tmpdir); err != nil {
+			log.Error("unable to clean up tmpdir", tmpdir, err)
+		}
+	}()
+
+	containerFSPath := path.Join(tmpdir, "fs")
+	if err := os.Mkdir(containerFSPath, 0755); err != nil {
+		return fmt.Errorf("%w: %s: %s", errors.ErrCreateTempDir, containerFSPath, err)
+	}
+
+	// export/flatten, and extract
+	log.Debug("exporting and flattening image")
+	r, w := io.Pipe()
+	go func() {
+		defer w.Close()
+		log.Debug("writing container filesystem to output dir", containerFSPath)
+		err = crane.Export(img, w)
+		if err != nil {
+			// TODO: Handle this error more effectively. Right now we rely on
```

Comment:
The [Export](https://github.com/google/go-containerregistry/blob/de8aff85c9090d37ed6fe436d696f3a820e5c767/pkg/crane/export.go#L26) function is *really* simple. In fact, the error that is returned would be related to the Writer. So, we probably don't even care about it here. I feel like we could actually just call crane.Export(), ignore the err, and w.Close(), and exit the goroutine. Any error would then be reported on the Reader side, so as long as the errors are handled there (which are with the Untar func), that should be sufficient.

Reply:
@bcrochet Understood. Would the action here be to remove the comment, or remove the logging? The logging seems beneficial, even if it's just one more breadcrumb telling us that it was some issue in the export vs. an issue in the extraction.

What do you think?